Actions/AL-Go-Helper.ps1

```diff
@@ -258,70 +280,106 @@ function DownloadAndImportBcContainerHelper {
             $params += @{ "bcContainerHelperConfigFile" = $repoSettingsPath }
         }
     }
-    if ($bcContainerHelperVersion -eq "") {
-        $bcContainerHelperVersion = "latest"
-    }
-    elseif ($bcContainerHelperVersion -eq "private") {
-        # Using a private BcContainerHelper version grabs a fork of BcContainerHelper with the same owner as the AL-Go actions
-        # The ActionsRepo below will be modified to point to actual running actions repo by the deploy mechanism, please do not change
-        $ActionsRepo = "microsoft/AL-Go-Actions@main"
-        $owner = $actionsRepo.Split('/')[0]
-        $bcContainerHelperVersion = "https://github.com/$owner/navcontainerhelper/archive/master.zip"
-    }
 
-    if ($bcContainerHelperVersion -eq "none") {
-        $tempName = ""
+    if ("$env:BcContainerHelperPath" -and (Test-Path -Path $env:BcContainerHelperPath -PathType Leaf)) {
+        $bcContainerHelperPath = $env:BcContainerHelperPath
+    }
+    elseif ($bcContainerHelperVersion -eq "none") {
         $module = Get-Module BcContainerHelper
         if (-not $module) {
             OutputError "When setting BcContainerHelperVersion to none, you need to ensure that BcContainerHelper is installed on the build agent"
         }
-
-        $BcContainerHelperPath = Join-Path (Split-Path $module.Path -parent) "BcContainerHelper.ps1" -Resolve
+        $bcContainerHelperPath = Join-Path (Split-Path $module.Path -parent) "BcContainerHelper.ps1" -Resolve
     }
     else {
-        $tempName = Join-Path ([System.IO.Path]::GetTempPath()) ([Guid]::NewGuid().ToString())
+        if ($bcContainerHelperVersion -eq "") {
+            $bcContainerHelperVersion = "latest"
+        }
+        elseif ($bcContainerHelperVersion -eq "private") {
+            # Using a private BcContainerHelper version grabs a fork of BcContainerHelper with the same owner as the AL-Go actions
+            # The ActionsRepo below will be modified to point to actual running actions repo by the deploy mechanism, please do not change
+            $ActionsRepo = "microsoft/AL-Go-Actions@main"
```

Comment:
I'm not sure I understand this. I don't see $ActionsRepo and $owner used later on in this function. Could we just hardcode $bcContainerHelperVersion = "https://github.com/microsoft/navcontainerhelper/archive/master.zip"?

Reply:
The reason for this one is really only when deploying to three repos and using your own ContainerHelper, like:
![image](https://github.com/microsoft/AL-Go/assets/10775043/7e67abcd-b2c1-469c-a028-523593905ba7)

It isn't really used anymore when developing with direct AL-Go development mechanism and I will consider whether we can remove this.

Reply:
I have replaced this with an error instead